app/src/gui/components/record/relationships/field_level_links/datagrid.tsx

```diff
@@ -150,6 +158,17 @@ export default function DataGridFieldLinksComponent(
     null as null | GridRowParams['row']
   );
   const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [uiSpec, setUISpec] = React.useState<ProjectUIModel | null>(null);
+
+  useEffect(() => {
+    const get = async () => {
+      const u = await getUiSpecForProject(props.project_id);
+      setUISpec(u);
+    };
+
+    get();
+  }, [props.project_id, props.field_name]);
```

Comment:
I'm a bit worried about performance/complexity. Shouldn't the ui spec be available already at this point given we are deep within a form? Why does it need to fetch it again?

Reply:
I agree, we should have a context provider or something.